packages/repo-tools/src/commands/openapi/client/generate.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import chalk from 'chalk';
+import { resolve } from 'path';
+import { OPENAPI_IGNORE_FILES, OUTPUT_PATH } from '../constants';
+import { promisify } from 'util';
+import { exec as execCb } from 'child_process';
+import { paths as cliPaths } from '../../../lib/paths';
+import { mkdirpSync } from 'fs-extra';
+import fs from 'fs-extra';
+
+const exec = promisify(execCb);
+
+async function generate(spec: string, outputDirectory: string) {
+  const resolvedOpenapiPath = resolve(spec);
+  const resolvedOutputDirectory = resolve(outputDirectory, OUTPUT_PATH);
+  mkdirpSync(resolvedOutputDirectory);
+
+  await fs.mkdirp(resolvedOutputDirectory);
+
+  await fs.writeFile(
+    resolve(resolvedOutputDirectory, '.openapi-generator-ignore'),
+    OPENAPI_IGNORE_FILES.join('\n'),
+  );
+
+  let opticLocation = '';
+  try {
+    opticLocation = (
+      await exec(`yarn bin openapi-generator-cli`, { cwd: cliPaths.ownRoot })
+    ).stdout as string;
+  } catch (err) {
+    throw new Error(
+      `Failed to find an openapi-generator-cli installation, ensure that you have openapi-generator-cli installed in the root of your repo. If not, run yarn add openapi-generator-cli from the root of your repo.`,
+    );
+  }
+
+  await exec(
+    `yarn openapi-generator-cli generate -i "${resolvedOpenapiPath}" -o "${resolvedOutputDirectory}" -g typescript -c "templates/typescript-backstage.yaml"`,
```

Comment:
rework to use above location var. execFile wasn't working as expected here and will need more tweaking.